deps/curl-7.88.1/lib/vtls/sf_ocsp.c

```diff
@@ -266,6 +268,53 @@ static const unsigned int MAX_SLEEP_TIME = 16000000;
 static const int MAX_RETRY = 1;
 
 
+/* Encode request data as a part of URL
+ * Encode all characters reserved by URL
+ * @param url_data The source data needs to be encoded_ptr
+ * @param data_size The size of the source data
+ * @param encoded_ptr The allocated buffer filled with encoded data, need to be freed by caller
+ * return the size of encoded data
+ */
+size_t encodeUrlData(const char *url_data, size_t data_size, char** encoded_ptr)
+{
+  // allocate buffer 3 times larger than source data in case every single, add 1 for '\0'
+  // character needs to be encoded as %xx
+  size_t buf_len = data_size * 3 + 1;
+  char* encode_buf = (char*)malloc(buf_len);
+  char* cur_ptr = encode_buf;
+  size_t enc_len = 0;
+  size_t pos = 0;
+
+  // encode all special characters
+  for (pos = 0; pos < data_size; pos++)
+  {
+    // if unreserved, put as is
+    // RFC 3986 section 2.3 Unreserved Characters
+    char car = url_data[pos];
+    if ((car >= '0' && car <= '9') ||
+        (car >= 'A' && car <= 'Z') ||
+        (car >= 'a' && car <= 'z') ||
+        (car == '-' || car == '_' || car == '.' || car == '~'))
+    {
+      *cur_ptr = car;
+      cur_ptr++;
+      enc_len++;
+      buf_len--;
+    }
+    else
+    {
+      snprintf(cur_ptr, buf_len, "%%%.2X", car);
+      cur_ptr += 3;
+      enc_len += 3;
+      buf_len -= 3;
+    }
+  }
+
+  *cur_ptr = '\0';
+  *encoded_ptr = encode_buf;
+  return enc_len;
```

Comment:
Are we using in the caller function?

Reply:
Not really but won't hurt to have data length returned. Make it optional output argument.